csharp/ql/test/shared/FlowSummaries.qll

```diff
@@ -0,0 +1,38 @@
+import semmle.code.csharp.dataflow.FlowSummary
+import semmle.code.csharp.dataflow.internal.FlowSummaryImpl::Private::TestOutput
+
+abstract class IncludeSummarizedCallable extends RelevantSummarizedCallable {
+  /** Gets the qualified parameter types of this callable as a comma-separated string */
+  private string parameterQualifiedTypeNamesToString() {
+    result =
+      concat(int i |
+        exists(this.getParameter(i))
+      |
+        this.getParameter(i).getType().getQualifiedName(), ", " order by i
+      )
```

Comment:
I think this can be made a bit nicer with
```suggestion
    result =
      concat(Parameter p, int i |
        p = this.getParameter(i)
      |
        p.getType().getQualifiedName(), ", " order by i
      )
```

Reply:
Yes, agree - this is nicer!